controllers/storageclusterpeer/storageclusterpeer_controller.go

```diff
@@ -18,46 +18,520 @@ package storageclusterpeer
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the StorageClusterPeer object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
-func (r *StorageClusterPeerReconciler) Reconcile(ctx context.Context, _ ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+func (r *StorageClusterPeerReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	var err error
+	r.ctx = ctx
+	r.log = log.FromContext(ctx, "StorageClient", request)
+	r.log.Info("Reconciling StorageClusterPeer.")
+
+	// Fetch the StorageClusterPeer instance
+	r.storageClusterPeer = &v1.StorageClusterPeer{}
+	r.storageClusterPeer.Name = request.Name
+	r.storageClusterPeer.Namespace = request.Namespace
+
+	if err = r.get(r.storageClusterPeer); err != nil {
+		if errors.IsNotFound(err) {
+			r.log.Info("StorageClusterPeer resource not found. Ignoring since object must be deleted.")
+			return reconcile.Result{}, nil
+		}
+		r.log.Error(err, "Failed to get StorageClusterPeer.")
+		return reconcile.Result{}, err
+	}
 
-	// TODO(user): your logic here
+	// Fetch local StorageCluster
+	r.storageCluster = &v1.StorageCluster{}
+	r.storageCluster.Name = r.storageClusterPeer.Spec.LocalCluster.Name.Name
+	r.storageCluster.Namespace = r.storageClusterPeer.Namespace
+	if err = r.get(r.storageCluster); err != nil {
+		if errors.IsNotFound(err) {
+			r.log.Info("StorageCluster resource not found. Stopping Reconcile since object must be deleted.")
+			return reconcile.Result{}, nil
+		}
+		r.log.Error(err, "Failed to get StorageCluster.")
+		return reconcile.Result{}, err
+	}
 
-	return ctrl.Result{}, nil
+	r.remoteStorageClusterName = types.NamespacedName{
+		Name:      r.storageClusterPeer.Spec.RemoteCluster.StorageClusterName.Name,
+		Namespace: r.storageClusterPeer.Spec.RemoteCluster.StorageClusterName.Namespace}
+
+	result, reconcileErr := r.reconcilePhases()
+
+	// Apply status changes to the StorageClient
+	statusErr := r.Client.Status().Update(ctx, r.storageClusterPeer)
+	if statusErr != nil {
+		r.log.Error(statusErr, "Failed to update StorageClusterPeer status.")
+	}
+	if reconcileErr != nil {
+		err = reconcileErr
+	} else if statusErr != nil {
+		err = statusErr
+	}
+	return result, err
 }
 
-// SetupWithManager sets up the controller with the Manager.
-func (r *StorageClusterPeerReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&v1.StorageClusterPeer{}).
-		Complete(r)
+func (r *StorageClusterPeerReconciler) reconcilePhases() (ctrl.Result, error) {
+	ocsClient, err := r.newExternalClusterClient()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	defer ocsClient.Close()
+
```

Comment:
the layout of this func seems a bit off, it is taking a decision around deletion at both top & inner level, possible to separate the deletion flow? Or did you try that already and hit issues?